### PR TITLE
feat: add lookup invoice to nwc webln provider

### PR DESCRIPTION
**Pull request by rolznz**
this is not actually a webln method. Not sure what to do here.

### CODE REVIEW

File: src/webln/NostrWeblnProvider.ts [hunk 1]
```diff
@@ -283,6 +294,19 @@ export class NostrWebLNProvider implements WebLNProvider, Nip07Provider {
       (result) => ({ paymentRequest: result.invoice }),
     );
   }
+
+  lookupInvoice(args: LookupInvoiceArgs) {
+    this.checkConnected();
+
+    return this.executeNip47Request<LookupInvoiceResponse>(
+      "lookup_invoice",
+      "lookupinvoice",
```
Comment (bumi) [inline src/webln/NostrWeblnProvider.ts:303]: do we actually need to pass the two here? or can this be some mapping? 
this is only needed for the events, isn't it?

File: src/webln/NostrWeblnProvider.ts [hunk 2]
```diff
@@ -214,7 +224,7 @@ export class NostrWebLNProvider implements WebLNProvider, Nip07Provider {
   // TODO: use NIP-47 get_info call
   async getInfo(): Promise<GetInfoResponse> {
     return {
-      methods: ["getInfo", "sendPayment", "addinvoice", "getBalance"],
+      methods: ["getInfo", "sendPayment", "addinvoice", "getBalance", "lookupinvoice"],
```
[the remark this answers is not in the document]
Comment (rolznz) [reply src/webln/NostrWeblnProvider.ts:227]: @bumi thanks for confirming, I got a bit confused here.

The ones I put are LND methods, but then we also have the official WebLN methods.

File: src/webln/NostrWeblnProvider.ts (hunk 1 above)
Comment (rolznz) [reply src/webln/NostrWeblnProvider.ts:303]: @bumi the webln request method is needed for the notification function. WebLN and NWC event names are inconsistent, I think we need to provide both.

Unless we can guarantee the only change is in casing, then we could have a mapping function from camelCase to lower_case_with_underscores

File: src/webln/NostrWeblnProvider.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rolznz) [reply src/webln/NostrWeblnProvider.ts]: @bumi I have updated webln-types and fixed those inconsistencies. Thanks!

File: src/webln/NostrWeblnProvider.ts (hunk 1 above)
Comment (bumi) [reply src/webln/NostrWeblnProvider.ts]: but can't we do a mapping from 

```
a = { 
  "lookup_invoice" : "lookupinvoice" 
}
a["lookup_invoice"]
```

that then can be checked?

**Pull request merged**